nuxeo-core/nuxeo-core-test/src/test/java/org/nuxeo/ecm/core/io/marshallers/json/types/FacetJsonWriterTest.java

```diff
@@ -53,14 +56,17 @@ public void testFacetWithoutSchema() throws Exception {
 
     @Test
     public void testFacetWithSchema() throws Exception {
-        CompositeType type = schemaManager.getFacet(HAS_RELATED_TEXT);
-        JsonAssert json = jsonAssert(type);
-        json.properties(3);
-        json.has("entity-type").isEquals("facet");
-        json.has("name").isEquals(HAS_RELATED_TEXT);
-        json = json.has("schemas").length(1).has(0);
-        json.has("entity-type").isEquals("schema");
-        json.has("name").isEquals("relatedtext");
+        var schemaByFacetName = Map.of(HAS_RELATED_TEXT, "relatedtext", COLD_STORAGE, "coldstorage");
+        for (Map.Entry<String, String> entry : schemaByFacetName.entrySet()) {
+            CompositeType type = schemaManager.getFacet(entry.getKey());
+            JsonAssert json = jsonAssert(type);
+            json.properties(3);
+            json.has("entity-type").isEquals("facet");
+            json.has("name").isEquals(entry.getKey());
+            json = json.has("schemas").length(1).has(0);
+            json.has("entity-type").isEquals("schema");
+            json.has("name").isEquals(entry.getValue());
+        }
```

Comment:
Not sure about this test, I think it tests that we can write a facet but not a specific one. I'm in favor of reverting change to not make this test more complex.